packages/flutter/test/material/reorderable_list_test.dart

```diff
@@ -655,6 +655,49 @@ void main() {
         expect(findState(const Key('A')).checked, true);
       });
 
+      testWidgets('Preserves children states across reorder when keys are not identical', (WidgetTester tester) async {
+        _StatefulState findState(Key key) {
+          return find.byElementPredicate((Element element) => element.ancestorWidgetOfExactType(_Stateful)?.key == key)
+              .evaluate()
+              .first
+              .ancestorStateOfType(const TypeMatcher<_StatefulState>());
+        }
+        await tester.pumpWidget(MaterialApp(
+          home: ReorderableListView(
+            children: <Widget>[
+              _Stateful(key: const ObjectKey('A')),
+              _Stateful(key: const ObjectKey('B')),
+              _Stateful(key: const ObjectKey('C')),
+            ],
+            onReorder: (int oldIndex, int newIndex) { },
+            scrollDirection: Axis.horizontal,
+          ),
+        ));
+        await tester.tap(find.byKey(const ObjectKey('A')));
+        await tester.pumpAndSettle();
+        // Only the 'A' widget should be checked.
+        expect(findState(const ObjectKey('A')).checked, true);
+        expect(findState(const ObjectKey('B')).checked, false);
+        expect(findState(const ObjectKey('C')).checked, false);
+
+        // Rebuild with distinct key objects.
+        await tester.pumpWidget(MaterialApp(
+          home: ReorderableListView(
+            children: <Widget>[
+              _Stateful(key: ObjectKey('B')), // ignore:prefer_const_constructors
```

Comment:
Please leave a comment with all `// ignore:` explaining why this has to be ignored as per our style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#comment-all-ignores

Reply:
The comment a few lines above ("Rebuild with distinct key objects") is meant to be that comment. I can make it more explicit if it's not clear.

Alternatively, I could create some other key type that doesn't have a const constructor at all, but it seemed a bit heavyweight for this.